src/Chapter3.hs

```diff
@@ -374,6 +380,27 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = MkKnight 
+  { knightHealth :: Int
+  , knightAttack :: Int
+  , knightGold :: Int
+  }
+
+data Monster = MkMonster 
+  { monsterHealth :: Int
+  , monsterAttack :: Int
+  , monsterGold :: Int
+  }
+
+fight kn mn = if knightWins (knightHealth kn) (monsterHealth mn) (knightAttack kn) (monsterAttack mn) 
```

Comment:
It's always a good idea to add top-level type signatures. And since you're working with custom data types, it's a good practice as well :slightly_smiling_face: